File: app/src/main/java/org/jellyfin/androidtv/constant/HomeSectionType.java
package org.jellyfin.androidtv.constant;

// Names used in jellyfin-web
// Only uses those that are configurable in web settings
public enum HomeSectionType {
    LATEST_MEDIA("latestmedia"),
    LIBRARY_TILES_SMALL("smalllibrarytiles"),
    LIBRARY_BUTTONS("librarybuttons"),
    RESUME("resume"),
    RESUME_AUDIO("resumeaudio"),
    ACTIVE_RECORDINGS("activerecordings"),
    NEXT_UP("nextup"),
    LIVE_TV("livetv"),

    NONE("none");

    public static HomeSectionType getByName(String name) {
        for (HomeSectionType type : HomeSectionType.values()) {
            if (type.name.equalsIgnoreCase(name)) return type;
        }

        return null;
    }

    public final String name;

    HomeSectionType(String name) {
        this.name = name;
    }
}

File: app/src/main/java/org/jellyfin/androidtv/constant/HomeSectionType.kt
package org.jellyfin.androidtv.constant

import org.jellyfin.androidtv.R
import org.jellyfin.preference.PreferenceEnum

/**
 * All possible homesections, "synced" with jellyfin-web.
 *
 * https://github.com/jellyfin/jellyfin-web/blob/master/src/components/homesections/homesections.js
 */
enum class HomeSectionType(
	override val serializedName: String,
	override val nameRes: Int,
) : PreferenceEnum {
	LATEST_MEDIA("latestmedia", R.string.home_section_latest_media),
	LIBRARY_TILES_SMALL("smalllibrarytiles", R.string.home_section_library),
	LIBRARY_BUTTONS("librarybuttons", R.string.home_section_library_small),
	RESUME("resume", R.string.home_section_resume),
	RESUME_AUDIO("resumeaudio", R.string.home_section_resume_audio),
	RESUME_BOOK("resumebook", R.string.home_section_resume_book),
	ACTIVE_RECORDINGS("activerecordings", R.string.home_section_active_recordings),
	NEXT_UP("nextup", R.string.home_section_next_up),
	LIVE_TV("livetv", R.string.home_section_livetv),
	RECENTLY_RELEASED("recentlyreleased", R.string.home_section_recently_released),
	RECENTLY_RELEASED_ADDED("recentlyreleasedadded", R.string.home_section_recently_released_added),
	RECENTLY_ADDED_RELEASED("recentlyaddedrealeased", R.string.home_section_recently_added_released),
	NONE("none", R.string.home_section_none),
}
